tests/acceptance/page_objects/organization_integration_settings.py

```diff
@@ -0,0 +1,129 @@
+        return self.element.find_elements_by_css_selector(
+            self.integration_name_selector
+        )
+
+    def get_installation_with_name(self, name):
+        for installation in self.installations:
+            if installation.get_attribute("innerText") == name:
+                return installation
+        return None
+
+
+class InstallationElement(BaseElement):
+    configure_button_selector = '[data-testid="integration-configure-button"]'
+    remove_button_selector = '[data-testid="integration-remove-button"]'
+
+    def __init__(self, integration, *args, **kwargs):
+        super(InstallationElement, self).__init__(
+            *args, **kwargs
+        )
+        self.integration = integration
+        self.configure_button = ButtonWithIconElement(
+            element=self.element.find_element_by_css_selector(self.configure_button_selector),
+        )
+        self.remove_button = ButtonWithIconElement(
+            element=self.element.find_element_by_css_selector(self.remove_button_selector)
+        )
+
+
+class IntegrationDetailsModal(ModalElement):
+    title_selector = '[data-testid="provider-name"]'
+    cancel_button_selector = '[data-testid="cancel-button"]'
+    add_button_selector = '[data-testid="add-button"]'
+
+    def __init__(self, provider, *args, **kwargs):
+        super(IntegrationDetailsModal, self).__init__(*args, **kwargs)
+        self.cancel_button = ButtonElement(
+            element=self.element.find_element_by_css_selector(self.cancel_button_selector),
+        )
+        self.add_button = ButtonElement(
+            element=self.element.find_element_by_css_selector(self.add_button_selector),
+        )
+        self.provider = provider
+
+    @property
+    def title(self):
+        return self.element.find_element_by_css_selector(
+            self.title_selector).get_attribute("innerText")
+
+
+class IntegrationSetupWindowElement(ModalElement):
+    title_selector = ''
+
+    def fill_in_setup_form(self, installation_data):
+        pass
+
+
+class ExampleIntegrationSetupWindowElement(IntegrationSetupWindowElement):
+    name_field_selector = 'name'
+    submit_button_selector = '[type="submit"]'
+
+    def __init__(self, *args, **kwargs):
+        super(ExampleIntegrationSetupWindowElement, self).__init__(*args, **kwargs)
+        self.name = self.element.find_element_by_name('name')
+        continue_button_element = self.element.find_element_by_css_selector(
+            self.submit_button_selector)
+        self.continue_button = ButtonElement(continue_button_element)
+
+    def fill_in_setup_form(self, installation_data):
+        self.name.send_keys(installation_data[self.name_field_selector])
+
+
+class OrganizationIntegrationSettingsPage(BasePage):
+    modal_selector = '.modal-dialog'
+
+    def __init__(self, providers=None, *args, **kwargs):
+        super(OrganizationIntegrationSettingsPage, self).__init__(*args, **kwargs)
+
+    def assert_correct_page(self):
```

Comment:
@markstory We talked about this briefly last week, but I was wondering how I could make this more useful. Would asserting that particular css elements are on the page? (i.e. assert the integration provider's table is there and assert the different providers on the page?)

Reply:
I could see that kind of is the page shaped right being a test method. I think if we're interested in check that the looks right we should probably be using percy.